backend/src/lib.rs

```diff
@@ -107,15 +107,7 @@ impl BackendType {
                 Box::new(composite::CompositeBackendFactory::new(plonky3::Factory))
             }
             #[cfg(feature = "stwo")]
-            BackendType::Stwo => Box::new(stwo::StwoProverFactory),
-            #[cfg(not(any(
-                feature = "halo2",
-                feature = "estark-polygon",
-                feature = "estark-starky",
-                feature = "plonky3",
-                feature = "stwo"
-            )))]
-            _ => panic!("Empty backend."),
```

Comment:
This should probably stay

Reply:
why was it ok to remove this?

Reply:
It seems somehow at least one backend feature will always be enabled (I don't know why, maybe it is related to default features being "default = ["halo2", "plonky3"]" in powdr/Cargo.toml, or defalut=[plonky3] in cli/Cargo.toml?). Then this code become redundant. At one point this code gave me error (I forgot what error it is). Removing it seems fine so it ends up here

Reply:
Yes but that only applies if you're compiling the backend crate together with these other crates, if you try to compile only the backend crate it will fail, which I think we should not allow